### PR TITLE
remove Dockerfile and .dockerignore for Java connectors

### LINKED ISSUE #30708: Remove airbyteDocker and all java dockerfiles for connectors
## Problem
Its confusing that
1. we have dockerfiles
2. and airbyteDocker
3. but dont use them for building and testing any more
4. 

## Solution
Remove them.

[slack conversation](https://airbytehq-team.slack.com/archives/C03VDJ4FMJB/p1695417277750759)

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Closes #30708
A clumsy merge reintroduced some java connnectors Dockerfile.
And we also forgot to remove `.dockerignore` files

## How
Remove `Dockerfile` and `.dockerignore` files for Java connectors.
Replace `airbyteDocker` in README with `buildConnectorImage`

Java connectors docker images are 💯 built via `airbyte-ci` and don't require Dockerfile anymore.

### CODE REVIEW

**Comment by alafanechere** [comment]
Current dependencies on/for this PR:
* master
  * **PR #31769** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/31769" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/31769?utm_source=stack-comment).

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/destination-bigquery/README.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Note that the `secrets` directory is git-ignored by default, so there is no dang
 
 #### Build
 Build the connector image via Gradle:
+You need a local install of [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) to run this command.
```
Comment (postamar) [inline airbyte-integrations/connectors/destination-bigquery/README.md:19]: You don't. Gradle will provision a python venv and do the poetry song-and-dance to install airbyte-ci from source in the .venv folder in the repo root.

**Review by postamar — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/destination-bigquery/README.md (hunk 1 above)
Comment (alafanechere) [reply airbyte-integrations/connectors/destination-bigquery/README.md:19]: Oh yes, I forgot your wizardries!

**Review by postamar — APPROVED** [review]

**Comment by octavia-approvington** [comment]
You did it! <br/> finish line

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**